### ISSUE TITLE
No resize event when going fullscreen

### ISSUE THREAD

**Issue by jube** [SFML/SFML#893]
No resize event is sent when the window is going fullscreen. Calling setSize() in fullscreen does not trigger any event either. I work on a Linux system (Debian unstable).

When pressing F, the following code only displays "Resized: 200x200".

```cpp
#include <cstdio>
#include <SFML/Graphics.hpp>

int main() {
  bool fullscreen = false;

  sf::VideoMode initialMode(200, 200);

  sf::RenderWindow window(initialMode, "Bug");

  while (window.isOpen()) {
    sf::Event event;

    while (window.pollEvent(event)) {
      switch (event.type) {
        case sf::Event::Closed:
          window.close();
          break;

        case sf::Event::Resized:
          std::printf("Resized: %ux%u\n", event.size.width, event.size.height);
          break;

        case sf::Event::KeyPressed:
          if (event.key.code == sf::Keyboard::F) {
            fullscreen = !fullscreen;
            sf::Uint32 style = sf::Style::Default;
            sf::VideoMode mode = initialMode;

            if (fullscreen) {
              style |= sf::Style::Fullscreen;
              mode = sf::VideoMode::getFullscreenModes().front();
            }

            window.create(mode, "Bug", style);
          }

        default:
          break;
      }
    }

    window.clear(sf::Color::White);

    window.display();
  }

  return 0;
}
```

**Comment by eXpl0it3r** [comment]
Thanks for the report. This has already been reported (#878) and is fixed in the [2.3.x branch](https://github.com/SFML/SFML/tree/2.3.x), meaning it will be included in the 2.3.1 patch release ones it gets out.

**Comment by jube** [comment]
On the 2.3.x branch, the previous code works. I changed it a little bit: I disabled key repeat and added a printf when F is pressed. It now displays:

```
Resized: 200x200                                                                                                                                                                       
Ask for fullscreen                                                                                                                                                                     
Resized: 1306x748                                                                                                                                                                      
Resized: 1366x768                                                                                                                                                                      
Ask for fullscreen                                                                                                                                                                     
Resized: 200x200
```
The "Resized: 1306x748" is weird (as if the window was maximized before going fullscreen) but the overall behaviour seems ok though.
